src/optimizing/Analyze_pattern.ml

```diff
@@ -85,14 +89,35 @@ let extract_strings_and_mvars ?lang any =
          * We do now semantic equivance on integers between values so
          * 1000 is now equivalent to 1_000 so we can't "regexpize" it.
          *)
-        | L (String (_, (str, _tok), _)) ->
+        | Call
+            ( { e = IdSpecial (Require, _); _ },
+              (_, [ Arg { e = L (String (_, (str, _tok), _)); _ } ], _) ) ->
             if not (Pattern.is_special_string_literal str) then
               Common.push str strings
         | IdSpecial (Eval, t) ->
             if Tok.is_origintok t then
               Common.push (Tok.content_of_tok t) strings
-        (* do not recurse there, the type does not have to be in the source *)
-        | TypedMetavar _ -> ()
+        | New (t, _, _, _) when lang = Some Lang.Java ->
+            (* We expect the 'new' keyword to be present.  *)
```

Comment:
meh, not super useful I think, most Java programs use 'new'.
In the same way if your pattern contains if/then/else/... we usually don't extract the strings for them because
they are too common.

But maybe it helps, I didn't run any benchmark.

Reply:
TBH I don't think I benchmarked it, but I'll do now.